dask/array/core.py

```diff
@@ -2079,7 +2088,226 @@ def unify_chunks(*args, **kwargs):
     return chunkss, arrays
 
 
+def _gencode(x, dsk, names, namespace, cache):
+    if core.istask(x):
+        func, args = x[0], x[1:]
+        kwargs = {}
+
+        if func is apply:
+            if len(args) == 3:
+                func, args, kwargs = args
+            else:
+                func, args = args
+
+        if func is partial_by_order:
+            if len(kwargs) > 2:
+                kwargs = kwargs.copy()
+                func, other = kwargs.pop('function'), kwargs.pop('other')
+            else:
+                func, other = kwargs['function'], kwargs['other']
+                kwargs = {}
+            args = list(args)
+            for i, arg in other:
+                args.insert(i, arg)
+
+        func = _gencode(func, dsk, names, namespace, cache)
+        args = [_gencode(a, dsk, names, namespace, cache) for a in args]
+        if kwargs:
+            args.append('**(%s)' % _gencode(kwargs, dsk, names, namespace, cache))
+        return "%s(%s)" % (func, ', '.join(args))
+    elif isinstance(x, list):
+        args = [_gencode(a, dsk, names, namespace, cache) for a in x]
+        return "[%s]" % ', '.join(args)
+    # Symbol
+    if core.ishashable(x):
+        if x in cache:
+            sym = cache[x]
+        else:
+            cache[x] = sym = next(names)
+            if x not in dsk:
+                namespace[sym] = x
+    else:
+        sym = next(names)
+        namespace[sym] = x
+
+    return sym
```

Comment:
I have a (probably bad) habit of writing a script to test things interactively, and then developing by writing code and running the test as `ipython -i --pdb test.py`. I find this easier to determine the source of bugs than with failures inside `pytest`, but as the "test" isn't written as a test I don't end up adding it in until later.